lib/DB/Utils.pm

```diff
@@ -189,4 +193,55 @@ sub updatePermissions ($ww3_conf, $role_perm_file) {
 	return;
 }
 
+=pod
+=head2 convertTimeDuration
+
+This subroutine converts time durations stored as a string in human-readable format
+to a number of seconds.
+
+=cut
+
+sub convertTimeDuration ($time_duration) {
+	if ($time_duration =~ /^(\d+)\s(sec)s?$/) {
+		return $1;
+	} elsif ($time_duration =~ /^(\d+)\s(min(ute)?)s?$/) {
+		return $1 * 60;
+	} elsif ($time_duration =~ /^(\d+)\s(h(ou)?r)s?$/) {
+		return $1 * 60 * 60;
+	} elsif ($time_duration =~ /^(\d+)\s(day)s?$/) {
+		return $1 * 60 * 60 * 24;
+	} elsif ($time_duration =~ /^(\d+)\s(week)s?$/) {
+		return $1 * 60 * 60 * 24 * 7;
+	} else {
+		return 0;
+	}
+}
+
+=pod
+=head2
+
+This coverts a number of seconds to a human-readable format
+
+=cut
+
+sub humanReadableTimeDuration ($td) {
+	my $times = {
+		week => int($td / 604800),
+		day  => int($td % 604800 / 86400),
+		hour => int($td % 86400 / 3600),
+		min  => int($td % 3600 / 60),
+		sec  => $td % 60
+	};
+
+	my $time_duration = '';
+	# Order is important so the keys are defined.
+	for (qw/week day hour min sec/) {
+		my $val = $times->{$_};
+		$time_duration .= ($time_duration ne '' && $val ? ', ' : '')
+			# pluralize for more than 1.
+			. ($val > 0 ? "$val $_" . ($val == 1 ? '' : 's') : '');
+	}
+	return $time_duration;
+}
```

Comment:
This is something that need to reconsidered even deeper.  What settings here are being saved in config files?  If a setting needs conversion to become readable then it probably is being done wrong.  Basically, it is probably incorrect to have a date at all in a config file.  Time durations can be purely in seconds (or minutes or whatever flat time unit) and still work perfectly well in a config file and be human readable.

Reply:
My thought with this is that time durations, it's easier to think in terms of minutes or days often instead of numbers of seconds (although clearly easy to calculate), so providing this functionality is relatively simple.  Again, we can chat more about this.